README.md

```diff
@@ -202,44 +255,111 @@ the `lookup` result will sometimes provide a new array, and sometimes not.
   supported.
 * **label** [String][]: Optional label in which to store the result.
 
-## `execute()`
+**Returns:** `this<SetupChain>`
 
-Executes each task added sequentially and collects the results into a single object.
-The results for `execute` are returned. The internal state will be Persisted.
+```javascript
+const SetupChain = require('@logdna/setup-chain')
 
-**returns**: [Object][]
+class MyChain extends SetupChain {
+  constructor(state) {
+    super(state, {
+      hello: async () => {
+        return 'hi'
+      }
+    })
+  }
+}
 
-```javascript
-await new SetupChain()
-  .foo()
-  .bar()
-  .insertDBRecord({}, 'record')
+const chain = new MyChain()
+chain
+  .repeat(5, 'hello', {}, 'result')
   .execute()
+  .then((result) => {
+    console.log(result)
+  })
 
-{
-  foo: 'bar'
-, bar: 'foo'
-, record: { id: 1 }
-}
+> {result: ['hi', 'hi', 'hi', 'hi', 'hi']}
 ```
 
-## `sleep(<Object>)`
+## `sleep(opts)`
+
+When used in a chain, it will wait for a specified number of milliseconds before returning. Since this is a chain action, the sleep will appear to happen sequentially
+within the action chain.
 
-### Parameters
+**Parameters**
 
 * **opts** [Object][]: configuration options
   * **ms** [Number][]: The number of milliseconds to wait before continuing
 
-This is a builder function for the `sleep` action.
-It will Waits for a specified number of milliseconds before returning.
+**Returns**: `this<SetupChain>`
 
-**returns**: SetupChain
+```javascript
+const chain = new SetupChain()
+const state = await chain
+  .set('start', '!date:getTime')
+  .sleep({ms: 1000})
+  .set('delayed_1', '!date:getTime')
+  .sleep({ms: 1000})
+  .set('delayed_2', '!date:getTime')
+  .execute()
+
+> {start: 1620496433509, delayed_1: 1620496434512, delayed_2: 1620496435514, set: undefined, sleep: undefined}
+```
 
 ## `reset()`
 
 Manually clears internal state and any pending tasks
 
-**returns**: SetupChain
+**Returns**: `this<SetupChain>`
+
+## `execute()`
+
+Resolves the promises returned by chain actions, and returns the result
+of each action in a single object. This should be the final
+method in any chain, otherwise the actions will not be resolved into values.
+After `execute` is called, the return value is also stored in `chain.state` for
+access later, even if the return value is not immediately used.
+
+**Returns**: [Object][]
+
+## `lookup(input)`
+
+Resolves lookup value. This function is generally used by actions to resolve values
+from the chain as it is executes, thus this function **is not chainable**. For a chainable action that persists a value to the chain state, use [set()](#setstringobject-any) instead.
```

Comment:
The problem with `lookup` is that its' the only thing that is good for calling things like `template`. That makes me think it's an action, but it doesn't return `this`.  That's what this paragraph is attempting to clear up.  Make a suggestion and I'll change it.

Reply:
> lookup and set would be opposing behaviors you me.

Ah, but in lots of cases, the result of `lookup` should be saved such that it can be used later in the chain, hence the need to use `set`.

I think all of this goes away if `lookup` was somehow an action as well.  The design of it doesn't feel right.  We've made it powerful enough to run functions, but no inherent way to save those results unless someone uses `set`

Reply:
`template` as an action seems ok, though I think we might be muddying the waters between actions + functions if we're exposing something like `template` as _both_.

in that case these would be equivalent, right?
```js
.set('foo', '!template:...')
.template('...', 'foo')
```

Are the real problems here that 1) `lookup` isn't chainable, and 2) the docs intermingling `set` here being in conflict? I agree with `set` being a non-obvious solution to wanting a `lookup` _on the chain_, but not sure about the `template` change in the context of this.

Reply:
they would be equivalent, but `set` feels like an ugly work around, and an explicit `template` action is really what you are after, and more obvious.

Reply:
Continuing this discussion outside of this PR and will do changes if needed later